utilities/transactions/pessimistic_transaction.cc

```diff
@@ -559,7 +561,18 @@ Status PessimisticTransaction::TryLock(ColumnFamilyHandle* column_family,
   // any writes since this transaction's snapshot.
   // TODO(agiardullo): could optimize by supporting shared txn locks in the
   // future
-  if (skip_validate || snapshot_ == nullptr) {
+  if (!do_validate || snapshot_ == nullptr) {
+    assert(!assume_exclusive_tracked || (previously_locked && !lock_upgrade));
+    if (assume_exclusive_tracked) {
+      if (!previously_locked) {
+        s = Status::InvalidArgument(
+            "assume_exclusive_tracked is set but it is not tracked yet");
+      } else if (lock_upgrade) {
+        s = Status::InvalidArgument(
```

Comment:
Discussed offline. Shared locking should be enough. The goal of validation is just to check whether writes have happened, and a shared lock would be able to prevent that.